rules/type-error.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const tcIdentifiers = new Set([
+	'isArguments',
+	'isArray',
+	'isArrayBuffer',
+	'isArrayLike',
+	'isArrayLikeObject',
+	'isBoolean',
+	'isBuffer',
+	'isDate',
+	'isElement',
+	'isEmptyObject',
+	'isError',
+	'isFinite',
+	'isFrozen',
```

Comment:
Thanks @jfmengels - i had to reread why i chose adding these (isSealed, too).Trying to change sealed/frozen Objects in a permitted way throws a `TypeError`, this is why i chose to treat isFrozen/isSealed-checks as a type check.

But this probably needs more discussion?

P.s.: A good [SO-answer](http://stackoverflow.com/questions/21402108/difference-between-freeze-and-seal-in-javascript) about sealed vs frozen.